[2025-07-18T14:56:30.355+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: restaurant_pipeline.send_to_kafka manual__2025-07-18T14:56:28.694717+00:00 [queued]>
[2025-07-18T14:56:30.361+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: restaurant_pipeline.send_to_kafka manual__2025-07-18T14:56:28.694717+00:00 [queued]>
[2025-07-18T14:56:30.361+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2025-07-18T14:56:30.368+0000] {taskinstance.py:1382} INFO - Executing <Task(BashOperator): send_to_kafka> on 2025-07-18 14:56:28.694717+00:00
[2025-07-18T14:56:30.371+0000] {standard_task_runner.py:57} INFO - Started process 277 to run task
[2025-07-18T14:56:30.374+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'restaurant_pipeline', 'send_to_kafka', 'manual__2025-07-18T14:56:28.694717+00:00', '--job-id', '36', '--raw', '--subdir', 'DAGS_FOLDER/restaurant_pipeline.py', '--cfg-path', '/tmp/tmps075rfag']
[2025-07-18T14:56:30.376+0000] {standard_task_runner.py:85} INFO - Job 36: Subtask send_to_kafka
[2025-07-18T14:56:30.406+0000] {task_command.py:416} INFO - Running <TaskInstance: restaurant_pipeline.send_to_kafka manual__2025-07-18T14:56:28.694717+00:00 [running]> on host 340b84e33f7d
[2025-07-18T14:56:30.451+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='moran' AIRFLOW_CTX_DAG_ID='restaurant_pipeline' AIRFLOW_CTX_TASK_ID='send_to_kafka' AIRFLOW_CTX_EXECUTION_DATE='2025-07-18T14:56:28.694717+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-18T14:56:28.694717+00:00'
[2025-07-18T14:56:30.452+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-07-18T14:56:30.453+0000] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', 'python /opt/streaming/producer.py']
[2025-07-18T14:56:30.460+0000] {subprocess.py:86} INFO - Output:
[2025-07-18T14:56:32.797+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-07-18T14:56:32.800+0000] {subprocess.py:93} INFO -   File "/opt/streaming/producer.py", line 7, in <module>
[2025-07-18T14:56:32.801+0000] {subprocess.py:93} INFO -     producer = KafkaProducer(
[2025-07-18T14:56:32.803+0000] {subprocess.py:93} INFO -   File "/home/***/.local/lib/python3.8/site-packages/kafka/producer/kafka.py", line 481, in __init__
[2025-07-18T14:56:32.806+0000] {subprocess.py:93} INFO -     client = self.config['kafka_client'](
[2025-07-18T14:56:32.807+0000] {subprocess.py:93} INFO -   File "/home/***/.local/lib/python3.8/site-packages/kafka/client_async.py", line 262, in __init__
[2025-07-18T14:56:32.807+0000] {subprocess.py:93} INFO -     self.config['api_version'] = self.check_version()
[2025-07-18T14:56:32.808+0000] {subprocess.py:93} INFO -   File "/home/***/.local/lib/python3.8/site-packages/kafka/client_async.py", line 1074, in check_version
[2025-07-18T14:56:32.809+0000] {subprocess.py:93} INFO -     raise Errors.NoBrokersAvailable()
[2025-07-18T14:56:32.810+0000] {subprocess.py:93} INFO - kafka.errors.NoBrokersAvailable: NoBrokersAvailable
[2025-07-18T14:56:32.845+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2025-07-18T14:56:32.853+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/bash.py", line 210, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-07-18T14:56:32.855+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=restaurant_pipeline, task_id=send_to_kafka, execution_date=20250718T145628, start_date=20250718T145630, end_date=20250718T145632
[2025-07-18T14:56:32.862+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 36 for task send_to_kafka (Bash command failed. The command returned a non-zero exit code 1.; 277)
[2025-07-18T14:56:32.885+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2025-07-18T14:56:32.912+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
